### PR TITLE
Login, Register, and EmailVerification support

### PR DESCRIPTION
**Pull request by TylerMathis**
## Describe your changes
There is quite a lot going on here, and I apologize for dumping this all in one PR. This PR consists of the following components.

| Login Screen | Register Screen | Email Verify Screen |
| - | - | - |
| Simulator Screen Shot - iPhone 14 Pro - 2023-01-12 at 17 31 14 | Simulator Screen Shot - iPhone 14 Pro - 2023-01-12 at 17 28 46 | Simulator Screen Shot - iPhone 14 Pro - 2023-01-12 at 17 28 57 |

These components look and work how you would expect. In order to achieve this, the "Tabs" Stack screen has been pulled out a layer in to `VerifyAuth`. This component handles all authorization changes, and renders the appropriate screen.

The views avoid keyboards, and Toasts are presented on errors such as server issues or invalid account sign ins.

## Issue ticket number and link
#45 

## Checklist before requesting a review
- [x] All code is linted and conforms to style guide
- [x] All necessary context is described in the PR or Issue
- [x] This branch is code and feature complete. If it is not complete, this PR is a draft.

### CODE REVIEW

File: utils/routes/root/paramList.ts [hunk 1]
```diff
@@ -1,8 +1,5 @@
-import { NativeStackScreenProps } from '@react-navigation/native-stack';
-import { ParamList as TabParamList } from '../tabs/paramList';
-
 export type ParamList = {
-  Tabs: NativeStackScreenProps<TabParamList>;
+  Tabs: undefined;
```
Comment (abrugal) [inline utils/routes/root/paramList.ts:2]: Not sure why this was changed to undefined, but I wrote the wrong thing either way, it should be NavigatorScreenParams\<TabParamList\>

File: utils/routes/tabs/paramList.ts [hunk 2]
```diff
@@ -4,8 +4,8 @@ import { ParamList as LeaderboardsParamList } from '../leaderboards/paramList';
 export type ParamList = {
   HomeTab: undefined;
   LeaderboardsTab: BottomTabScreenProps<LeaderboardsParamList>;
```
Comment (abrugal) [inline utils/routes/tabs/paramList.ts:6]: I also messed up here. Should be NavigatorScreenParams\<LeaderboardsParamList\>

File: utils/routes/home/routes.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ export type Route = {
 
 export const routes: Array<Route> = [
   {
-    name: 'Home',
+    name: 'MainFeed',
```
Comment (abrugal) [inline utils/routes/home/routes.tsx:12]: Is the plan to call it MainFeed? If so the name needs to be changed in the home/names and home/paramList

**Review by abrugal — CHANGES_REQUESTED** [review]
Left some comments. Other than that this looks awesome!

File: utils/routes/home/routes.tsx (hunk 3 above)
Comment (TylerMathis) [reply utils/routes/home/routes.tsx:12]: Oh I actually don't know why I changed this. Must have been on accident. Reverting now! Thanks for catching.

**Review by abrugal — APPROVED** [review]
Amazing PR

**Pull request merged**